app/perfsymboltable.cpp

```diff
@@ -40,14 +40,26 @@
 #include <debuginfod.h>
 #endif
 
-PerfSymbolTable::PerfSymbolTable(qint32 pid, Dwfl_Callbacks *callbacks, PerfUnwind *parent) :
-    m_perfMapFile(QDir::tempPath() + QDir::separator()
-                  + QLatin1String("perf-%1.map").arg(QString::number(pid))),
-    m_hasPerfMap(m_perfMapFile.exists()),
-    m_cacheIsDirty(false),
-    m_unwind(parent),
-    m_callbacks(callbacks),
-    m_pid(pid)
+namespace {
+QString perfMapFile(const QString& customPerfMapPath, int pid)
+{
+    if (!customPerfMapPath.isEmpty()) {
+        QString path = customPerfMapPath + QDir::separator() + QLatin1String("perf-%1.map").arg(QString::number(pid));
+        if (QFile::exists(path)) {
+            return path;
+        }
+    }
+    return QDir::tempPath() + QDir::separator() + QLatin1String("perf-%1.map").arg(QString::number(pid));
+}
+}
+
+PerfSymbolTable::PerfSymbolTable(qint32 pid, Dwfl_Callbacks* callbacks, const QString& perfMapPath, PerfUnwind* parent)
```

Comment:
instead of taking the QString perfMapPath as an arg, add a getter for that to PerfUnwind, then call it when you init the member variable below